BEP191.md

```diff
@@ -0,0 +1,204 @@
+
+## 2. Abstract
+
+This BEP describes a new slash mechanism to slash malicious validator in more scenarios, if some validator missing proposal, it could slash directly. Otherwise, validators could propose a slash to the malicious validator, and when the majority of validators agree on it, the slash will affect too.
+
+There are slash scenarios:
+
+1. The validator misses the `INITURN` block;
+2. The validator delay broadcast `INTURN` block too long;
+3. The validator proposes a duplicated block in the same block height with a different block hash;
+
+The new slash rules refer to the previous ones. There is a certain tolerance threshold, and the punishment will be executed if the threshold is exceeded. This BEP can also be used in other scenarios where validators are punished for malicious behavior.
+
+The new slash mechanism enriches the scenarios for monitoring validators, can promote good competition among validators, and protect the security and smooth operation of the network.
+
+## 3. Motivation
+
+The current BSC network will impose slash penalties on validators who skip block generation, but this is not enough. After observing that the current validator will delay the broadcast behavior:
+
+1. Poorly operated validator clients may use poor machines or network providers;
+2. Maliciously delay broadcasting blocks, in order to collect more transaction fees as rewards;
+
+These behaviors have seriously affected the stability of the network, but they will not be punished, because the validator's `INTURN` block in BSC's PoSA algorithm has a higher difficulty. In some scenarios, the slashing penalty can be reorg, which will damage the entire network. Not in line with the interests of the current BSC's long-term operations:
+
+1. Frequent network reorgs cannot guarantee a stable TPS;
+2. Collect more transactions during the delayed broadcast period to obtain higher returns, resulting in uneven returns and harming the rights and interests of other validators;
+
+In order to guarantee the rights and interests of the nodes and maintain the reliable and safe operation of the network, it is necessary to propose a solution, through an effective mechanism to limit the validator to package transactions for a longer period of time, or to use a poor network and machine to run the validator node.
+
+## 4. Specification
+
+The PoSA consensus algorithm is used in BSC, and the set of selected validators generates blocks in sequence according to the established order and broadcasts them to the network.
+
+The blocks released in the order of block generation are called `INTURN` blocks, and the block difficulty is 2. In order to ensure liveness, if there is a block node failure, severe network delay, or a malicious block, other validators will re-generate a liveness block. This block is called `NOTURN` Block, block difficulty is 1, and at this time the validator of the supposed block will mark a slash.
```

Comment:
`block node` seems a noun that very hard to understand.

Reply:
`This block is called `NOTURN` Block, block difficulty is 1, and at this time the validator of the supposed block will mark a slash.`. this does not always stand, need to check whether the supposed proposer ever propose a block in recent N/2 Blocks.